### PR TITLE
Elimination of R dependency, implementation of a new OOP module, cleaning install and testing scripts

### PR DESCRIPTION
**Pull request by celiosantosjr**
The new version of macrel does change some probabilities (14/899) in the testing due to the rounding problem when transferring from R to python. However, no classification changed and the behavior is stable. Now it uses one additional module (database) to perform the calculation of the same features done before and I rewrote this new module to implement more object-oriented programing.

The simple use of the modlamp was not possible because it uses different scales and to make it ordered I rewrote some functions and others were just adapted.

The elimination of R dependency was cleaned in the testing and installing routines which reduced the time of CI testing to 5 min.

### CODE REVIEW

**Comment by celiosantosjr** [comment]
All these commits will be squashed into a single message. I just left them to you,  then it would be possible to trace the changes.

**Comment by luispedro** [comment]
Thanks. We can change the results if we clearly mark it as a new version.

File: macrel/macrel_features.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+from .database import pos_pks, neg_pks, boman_scale
+from collections import Counter
+
+class MacrelFeatures:
+
+    '''
+    Functions were adapted from:
+
+      -- ModlAMP(https://github.com/alexarnimueller/modlAMP)
+      -- Peptides R Package (https://github.com/dosorio/Peptides), and
```
Comment (luispedro) [inline macrel/macrel_features.py:13]: Does this make the code GPL? How much adaptation was there?

File: macrel/AMP_features.py [hunk 2]
```diff
@@ -2,109 +2,40 @@
...
-                code.append((i + 1)/len(sequence) * 100)
-                break
-        else:
-            code.append(0)
-    return np.array(code)
+from Bio import SeqIO
+from .database import GROUPS_SA, GROUPS_HB
+from .macrel_features import MacrelFeatures
 
+def openfile(ifile):
```
Comment (luispedro) [inline macrel/AMP_features.py:14]: We already had a FASTA parsing function that didn't require Biopython (which is also a heavy dependency for just the FASTA parser)

File: macrel/AMP_features.py [hunk 3]
```diff
@@ -147,8 +78,8 @@ def main(args):
 
     ifile = args[1]
     ofile = args[2]
-    features = features(ifile)
```
Comment (luispedro) [inline macrel/AMP_features.py:150]: Why this change?

**Comment by luispedro** [comment]
Did you measure impact on performance? Replacing C++ by Python might be a big slowdown.

File: macrel/AMP_features.py (hunk 2 above)
Comment (celiosantosjr) [reply macrel/AMP_features.py:14]: I thought the same, but after the CLI testing, I checked that with the dependencies, Macrel package now is 200MB less heavy than before.

File: macrel/AMP_features.py (hunk 3 above)
Comment (celiosantosjr) [reply macrel/AMP_features.py:150]: just refactoring the variable name to reduce the chances of system confusion. In my computer when I call the code as a separate script, it returns that the variable was used before the assignment, which is pretty much solved by this small change.

File: macrel/macrel_features.py (hunk 1 above)
Comment (celiosantosjr) [reply macrel/macrel_features.py:13]: modlAMP is BSD3 which is free to share/modify/distribute just acknowledging to the authors. The code itself changed, however, some functions had very few modifications, and there are some functions that they did not have, such as hydrophobicity. I am still wondering if we should include the synthesis/solubility rules there and change the output a bit, giving also those info, what do you think?

**Comment by celiosantosjr** [comment]
The changes made actually slow down the program, but not so drastically. I tested with 1 million simulated peptides, to conclude the computation of the features in the modified version and the original version the difference was barely 15 min:

| **method / time** | **modified version** | **original** |
| :---: | :---: | :---: |
| real | 78m 16.521s | 63m 37.135s |
| user | 40m 30.502s | 40m 35.820s |
| sys | 0m 8.305s | 0m 15.105s |

**Comment by luispedro** [comment]
That is not so bad, actually; and could potentially be improved with a few tweaks (after merging)

**Comment by luispedro** [comment]
Can we move ahead with this? From my POV

1. To be rebased (maybe some manual edits will be necessary)
2. The GPL issue needs to be clarified
3. I still prefer the older `fasta.py` to introducing the `biopython` dependency and see no argument for that change (and if there is an argument for that change, maybe do it as a separate PR so these things can be discussed separately)
4. I am not a fan of the object-based interface (feels old-fashioned), but not a hill I will die on
